packages/react/src/lib/createComponent.tsx

```diff
@@ -22,7 +23,7 @@ export type CreateComponentReturnType<P> = React.FunctionComponent<P> & {
   create: Function
 }
 
-const createComponent = <P extends {} = {}, S extends {} = {}>({
+const createComponent = <P extends {} = ObjectOf<any>>({
```

Comment:
seems that the intent is a bit different here: if we would like to restrict type to objects, I would suggest the following expression for that:
```js
const createComponent = <P extends ObjectOf<any> = {}>({ ...
```

Reply:
the intent is to have `P` have `ObjectOf<any` by default so that we don't have the errors I mentioned in #946 for the case when we do not supply the props interface to `createComponent`

From #946 description:

### Actual Result
We have compilation error on our own example from documentation 💣
![screenshot 2019-02-22 at 12 25 30](https://user-images.githubusercontent.com/5442794/53239937-917a7d80-369d-11e9-8b12-b6ba8dc12620.png)
![screenshot 2019-02-22 at 12 25 44](https://user-images.githubusercontent.com/5442794/53239942-94756e00-369d-11e9-99a9-a061e3f26feb.png)

Reply:
yes - as a result, have agreed with @Bugaa92 to introduce the following type:

```js
const createComponent = <P extends ObjectOf<any> = {}>({ ..
```
This will prevent the following things:
```js
const Foo = createComponent<number>(...)
const Foo = createComponent<string>(...)
```
and, at the same time, will work properly for the scenarios @Bugaa92 has mentioned 👍

Reply:
after discussions with @kuzhelov we agreed to go with `P extends ObjectOf<any> = any`